### ISSUE TITLE
[FAILING TEST] Dangerous example code

### ISSUE THREAD

**Issue by jeffminsungkim** [jmcdo29/testing-nestjs#344]
**Describe the bug**
I'd like to point out that there an insecure code from cat.service.spect.ts file in `typeorm-sample` project. (Perhaps, there might be more..)

**To Reproduce**

```typescript
const catArray = [
  new Cat(testCat1, testBreed1, 4, 'uuid1'),
  new Cat('Test Cat 2', 'Test Breed 2', 3, 'uuid2'),
  new Cat('Test Cat 3', 'Test Breed 3', 2, 'uuid3'),
];

beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        CatService,
        {
          provide: getRepositoryToken(Cat),
          useValue: {
            find: jest.fn().mockResolvedValue(catArray),
          },
        },
      ],
    }).compile();
});

describe('getAll', () => {
  it('should return an array of cats', () => {
    expect(service.getAll()).resolves.toEqual([
      new Cat('Different', 'Value', 4, 'uuid1123213'),
      new Cat('Test Cat 2', 'Test Breed 2', 3, 'uuid2'),
      new Cat('Test Cat 3', 'Test Breed 3', 2, 'uuid3'),
    ]);
  });
});
```
It throws the following error below, but the test **still pass**.
`(node:40862) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)`

For some reason, if we change the value of `catArray`, the test code turns out to be pointless.

**Expected behavior**

```typescript
describe('getAll', () => {
  it('should return an array of cats', async () => {
    expect(await service.getAll()).toEqual([
      new Cat('Different', 'Value', 4, 'uuid1123213'),
      new Cat('Test Cat 2', 'Test Breed 2', 3, 'uuid2'),
      new Cat('Test Cat 3', 'Test Breed 3', 2, 'uuid3'),
    ]);
  });
});
```
Jay, if you think this needs to be fixed then I can get my hands dirty.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version: 6.11.8 (latetest)

**Additional context**

**Comment by jmcdo29** [comment]
Yeah! Go for it! Shouldn't be too hard of a PR to manage :smiley: I know we've talked about this before, would be great if this was fixed for better examples

**Comment by jmcdo29** [comment]
Fixed by #345
